app/src/main/java/org/cru/godtools/activity/GlobalDashboardActivity.kt

```diff
@@ -0,0 +1,41 @@
+package org.cru.godtools.activity
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import androidx.fragment.app.commit
+import org.cru.godtools.R
+import org.cru.godtools.analytics.model.AnalyticsScreenEvent
+import org.cru.godtools.analytics.model.AnalyticsScreenEvent.Companion.SCREEN_GLOBAL_DASHBOARD
+import org.cru.godtools.base.ui.activity.BaseActivity
+import org.cru.godtools.fragment.GlobalDashboardFragment
+
+fun Activity.startGlobalDashboardActivity() {
+    Intent(this, GlobalDashboardActivity::class.java)
+        .putExtras(BaseActivity.buildExtras(this))
+        .also { startActivity(it) }
+}
+
+class GlobalDashboardActivity : BasePlatformActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_generic_fragment)
+        loadGlobalDashboardFragment()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        mEventBus.post(AnalyticsScreenEvent(SCREEN_GLOBAL_DASHBOARD))
+    }
+
+    private fun loadGlobalDashboardFragment() {
+        val tag = "GlobalDashboardFragment"
+        with(supportFragmentManager) {
+            commit {
+                val fragment = findFragmentByTag(tag) ?: GlobalDashboardFragment()
+                replace(R.id.frame, fragment, tag).setPrimaryNavigationFragment(fragment)
```

Comment:
👎 
You don't want to do anything if the fragment is already attached, you should check for the fragments presence and not execute any transaction if the fragment exists.

also, use the fact that it's the primaryNavigationFragment to find it, there is no need to set a tag on the fragment